### PR TITLE
Preparing merge of PR #20

### LINKED ISSUE #18: A new configuration discovery strategy.
If I write an app which has code in `/opt/app` and I would like users to optionally put a `config.ini` in `/etc/app/`, there is no way to do it through prettyconf's current discovery strategy.
It would be nice to be able to configure this behavior.
I imagine something like:
```python
from prettyconf.configuration import env_vars, dirs, recursive

config.strategies = [
     env_vars, 
     dirs('/etc/app', '~/.configs/app'], 
     recursive(root_dir='/', starting_path='./')
]
```
The above snippet means that a config variable would be first checked within the environment variables,
if it is not found there, it would look at `/etc/app`, etc.

What do you think? I'm currently facing this issue and I'm willing to work on this.

### PR DESCRIPTION
**Pull request by osantana**
Closes #18

### CODE REVIEW

File: tests/test_filediscover.py [hunk 1]
```diff
@@ -6,9 +6,9 @@
 import shutil
 import tempfile
 
-from prettyconf.loaders import RecursiveSearch
 from prettyconf.exceptions import InvalidPath
 from prettyconf.loaders import IniFile
+from prettyconf.loaders import RecursiveSearch
```
Comment (hernantz) [inline tests/test_filediscover.py]: Here please do `from prettyconf.loaders import IniFile, RecursiveSearch`

File: .travis.yml [hunk 2]
```diff
@@ -4,36 +4,28 @@ sudo: false
...
       env:
-      - PYTHON_VERSION=3.4.3
+      - PYTHON_VERSION=3.5.6
       - PYENV_ROOT=~/.pyenv
       - PATH=$PYENV_ROOT/shims:$PYENV_ROOT/bin:$PATH
     - os: osx
       language: generic
       env:
-      - PYTHON_VERSION=3.5.0
+      - PYTHON_VERSION=3.6.8
```
Comment (hernantz) [inline .travis.yml]: Maybe we could also add python 3.7

Comment (osantana) [reply .travis.yml]: TravisCI does not provide 3.7 :(

Comment (scottwernervt) [reply .travis.yml]: You can use `dist: xenial` to get 3.7 support on travis, see https://github.com/scottwernervt/favicon/blob/master/.travis.yml#L15

**Comment by hernantz** [comment]
@osantana Looks good, I forgot to add `__repr__()` for the other loaders, I think we should add them for consistency.

**Comment by hernantz** [comment]
Throwing this out there, repr for `Configuration()` could be:
```python
    def __repr__(self):
        return 'Configuration(loaders=[{}])'.format(', '.join([repr(l) for l in self.loaders]))
```
which results in:

```python
>>> from prettyconf import config
>>> config
Configuration(loaders=[Environment(), RecursiveSearch(starting_path="/home/hernantz/devel/prettyconf/prettyconf", root_path="/")])
```

**Comment by osantana** [comment]
Let's go? :)

**Comment by hernantz** [comment]
It's a go on my end :)

**Comment by osantana** [comment]
Released! Thank you!
https://pypi.org/project/prettyconf/

**Pull request merged**